src/components/organisms/UiHorizontalPaging/UiHorizontalPaging.vue

```diff
@@ -286,6 +277,23 @@ const menuItemsSlots = computed(() => (Object.keys(slots).reduce((object, slotNa
   }
   return object;
 }, {})));
+const internalMenuTemplateRefs = ref<InstanceType<typeof UiMenu> | null>(null);
+watch(() => props.menuTemplateRefs, () => {
+  internalMenuTemplateRefs.value = props.menuTemplateRefs;
```

Comment:
Why can ref be both from parent and from this component?

Reply:
Because users can use the slot to replace the UiMenu template. An we need template refs to UiMenu.

Reply:
It looks a bit weird to me, both as watcher and replacing internals.
How about replacing the watcher with:
`const usableMenuTemplateRefs = computed(() => (props.menuTemplateRefs ?? internalMenuTemplateRefs)`?